sdk/identity/Azure.Identity/src/LinuxNativeMethods.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Azure.Identity
+{
+    internal static class LinuxNativeMethods
+    {
+        public enum SecretSchemaAttributeType
+        {
+            SECRET_SCHEMA_ATTRIBUTE_STRING = 0,
+            SECRET_SCHEMA_ATTRIBUTE_INTEGER = 1,
+            SECRET_SCHEMA_ATTRIBUTE_BOOLEAN = 2,
+        }
+
+        public enum SecretSchemaFlags
+        {
+            SECRET_SCHEMA_NONE = 0,
+            SECRET_SCHEMA_DONT_MATCH_NAME = 2,
+        }
+
+        internal struct GError
+        {
+            public uint Domain;
+            public int Code;
+            public string Message;
+        }
+
+        public static IntPtr secret_schema_new(string name, SecretSchemaFlags flags, string attribute1, SecretSchemaAttributeType attribute1Type, string attribute2, SecretSchemaAttributeType attribute2Type)
+        {
+            IntPtr namePtr = Marshal.StringToHGlobalAnsi(name);
+            IntPtr attribute1Ptr = Marshal.StringToHGlobalAnsi(attribute1);
+            IntPtr attribute2Ptr = Marshal.StringToHGlobalAnsi(attribute2);
+
+            try
+            {
+                return Imports.secret_schema_new(namePtr, (int)flags, attribute1Ptr, (int)attribute1Type, attribute2Ptr, (int)attribute2Type, IntPtr.Zero);
+            }
+            finally
+            {
+                Marshal.FreeHGlobal(attribute2Ptr);
+                Marshal.FreeHGlobal(attribute1Ptr);
+                Marshal.FreeHGlobal(namePtr);
+            }
+        }
+
+        public static string secret_password_lookup_sync(IntPtr schemaPtr, IntPtr cancellable, string attribute1Type, string attribute1Value, string attribute2Type, string attribute2Value)
+        {
+            IntPtr attribute1NamePtr = Marshal.StringToHGlobalAnsi(attribute1Type);
```

Comment:
Doesn't PInvoke do this when you mark parameter with `MarshalAs` ?

```
void PassString3([MarshalAs(UnmanagedType.LPStr)] string s);

```

Reply:
Needed `DllImport(CharSet = CharSet.Ansi, BestFitMapping = false, ThrowOnUnmappableChar = true)`